org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/GradleProjectImporter.java

```diff
@@ -100,37 +104,29 @@ private void importDir(Path rootFolder, IProgressMonitor monitor) {
 				}
 			}
 		}
-		startSynchronization(rootFolder.toFile(), distribution, NewProjectHandler.IMPORT_AND_MERGE);
+		return distribution;
 	}
 
-	protected void startSynchronization(File location, GradleDistribution distribution, NewProjectHandler newProjectHandler) {
-		List<String> jvmArgs = new ArrayList<>();
-		IProxyService proxyService = JavaLanguageServerPlugin.getInstance().getProxyService();
-		IProxyData proxy = proxyService.getProxyData(IProxyData.HTTP_PROXY_TYPE);
-		if (proxy != null) {
-			addArgs(jvmArgs, JavaLanguageServerPlugin.HTTP_PROXY_HOST, proxy.getHost());
-			if (proxy.getPort() > 0) {
-				addArgs(jvmArgs, JavaLanguageServerPlugin.HTTP_PROXY_PORT, String.valueOf(proxy.getPort()));
-			}
-			addArgs(jvmArgs, JavaLanguageServerPlugin.HTTP_PROXY_USER, proxy.getUserId());
-			addArgs(jvmArgs, JavaLanguageServerPlugin.HTTP_PROXY_PASSWORD, proxy.getPassword());
-		}
-		proxy = proxyService.getProxyData(IProxyData.HTTPS_PROXY_TYPE);
-		if (proxy != null) {
-			addArgs(jvmArgs, JavaLanguageServerPlugin.HTTPS_PROXY_HOST, proxy.getHost());
-			if (proxy.getPort() > 0) {
-				addArgs(jvmArgs, JavaLanguageServerPlugin.HTTPS_PROXY_PORT, String.valueOf(proxy.getPort()));
+	protected void startSynchronization(Path rootFolder, IProgressMonitor monitor) {
+		boolean shouldSynchronize = true;
+		File location = rootFolder.toFile();
+		IProject[] projects = ResourcesPlugin.getWorkspace().getRoot().getProjects();
+		for (IProject project : projects) {
+			File projectDir = project.getLocation() == null ? null : project.getLocation().toFile();
+			if (location.equals(projectDir)) {
+				PersistentModel model = CorePlugin.modelPersistence().loadModel(project);
+				if (model.isPresent()) {
+					shouldSynchronize = false;
+					break;
+				}
 			}
```

Comment:
```java
if (location.equals(projectDir)) {
    PersistentModel model = CorePlugin.modelPersistence().loadModel(project);
    if (model.isPresent()) {
        shouldSynchronize = false;
    }
    break;
}
```

Reply:
We mustn't do it when a model isn't present. See https://github.com/eclipse/buildship/blob/master/org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/workspace/internal/GradleClasspathContainerUpdater.java#L160

Reply:
sure, but if you found the project matching the location, regardless of its sync' state, there's no need iterating on other projects, since you shouldn't be able to find other projects with the same location (else that'd be a problem). Maybe separate the project lookup and the sync check if it makes it clearer.

Reply:
Agree.